shell/platform/android/io/flutter/plugin/editing/TextEditingDelta.java

```diff
@@ -0,0 +1,145 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugin.editing;
+
+import androidx.annotation.VisibleForTesting;
+import io.flutter.Log;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/// A representation of the change that occured to an editing state, along with the resulting
+/// composing and selection regions.
+public final class TextEditingDelta {
+  private CharSequence oldText;
+  private CharSequence deltaText;
+  private int deltaStart;
+  private int deltaEnd;
+  private int newSelectionStart;
+  private int newSelectionEnd;
+  private int newComposingStart;
+  private int newComposingEnd;
+
+  private static final String TAG = "TextEditingDelta";
+
+  public TextEditingDelta(
+      CharSequence oldEditable,
+      int replacementDestinationStart,
```

Comment:
Since in this system, we have a Delta type called `replacement`, we should avoid using the name replacement to describe two separate things. Can we not just use `delta<blah>` like you do in the constructor for testing below?

Also, what is the difference between that testing constructor and this one? They seem to do the same thing, but parameter naming is out of sync.

Reply:
The reason I don't call it `delta<blah>` is because the parameters are coming straight from Android `replace` method. The actual deltas are set by `setDeltas`.

Your right the testing constructor can be eliminated.